temba/ussd/models.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import absolute_import, unicode_literals
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.db import models
+from temba.channels.models import ChannelSession
+from temba.contacts.models import Contact, URN
+from temba.triggers.models import Trigger
+
+
+class USSDManager(models.Manager):
+    def get_queryset(self):
+        return super(USSDManager, self).get_queryset().filter(session_type=USSDSession.USSD)
+
+    def get_initiated_push_session(self, contact):
+        return self.get_queryset().filter(direction=USSDSession.USSD_PUSH, contact=contact).first()
+
+
+class USSDSession(ChannelSession):
+    USSD_PULL = INCOMING = 'I'
+    USSD_PUSH = OUTGOING = 'O'
+
+    objects = USSDManager()
+
+    class Meta:
+        proxy = True
+
+    def save(self, force_insert=False, force_update=False, using=None, update_fields=None):
+        if not self.pk:
+            self.session_type = USSDSession.USSD
+            user = User.objects.get(username=settings.ANONYMOUS_USER_NAME)
+            self.created_by = user
+            self.modified_by = user
```

Comment:
First I used the same manager create method as you did on IVR, but if you want to use update_or_create or get_or_create then that .create() is not getting called, thats why I moved it to save(). Now, I made a bit of a research and indeed you can't use Manager.create() for that, however there's a cool feature to [use a custom queryset as manager](https://docs.djangoproject.com/en/dev/topics/db/managers/#creating-a-manager-with-queryset-methods) . I tested it and it works that way.